app/models/spree/payment_method/adyen_credit_card.rb

```diff
@@ -14,11 +14,24 @@ def message
 
     include Spree::PaymentMethod::AdyenPaymentMethod
     preference :cse_library_location, :string
+    preference :cse_library_location_integrity, :string
 
     def cse_library_location
       ENV["ADYEN_CSE_LIBRARY_LOCATION"] || preferred_cse_library_location
     end
 
+    def cse_library_location_integrity
+      preferred_cse_library_location_integrity
+    end
+
+    def checkoutshopper_cse_library_location
+      return unless cse_library_location.present?
+
+      env = Rails.env.production? ? "live" : "test"
```

Comment:
Note that there is an existing `server` preference that takes the values "live" or "test". I wonder if it wouldn't be best to use that for testing in a live environment. However, I tested calling `preferred_server` on the method so perhaps it works differently.

Reply:
I'm not sure about that preference usage, so far `test` and `live` have been hardcoded in the CSE library URL preference, meaning that `preferred_server` didn't have any effect on them. But I think it's reasonable to use it instead of forcing an Adyen environment only depending on the Rails server. I think I'll change the code to reflect this, thanks.